adminSiteClient/VariablesAnnotationPage.tsx

```diff
@@ -0,0 +1,928 @@
+        )
+    else return undefined
+}
+
+@observer
+class VariablesAnnotationComponent extends React.Component {
+    static contextType = AdminAppContext
+
+    @observable.ref grapher = new Grapher() // the grapher instance we keep around and update
+    @observable.ref grapherElement?: JSX.Element // the JSX Element of the preview IF we want to display it currently
+    numTotalRows: number | undefined = undefined
+    selectedRow: number | undefined = undefined
+    @observable activeTab = Tabs.FilterTab
+
+    context!: AdminAppContextType
+    /** This array contains a description for every column, information like which field
+        to display, what editor control should be used, ... */
+    @observable.ref fieldDescriptions: FieldDescription[] | undefined =
+        undefined
+
+    /** Rows of the query result to the /variable-annotations endpoint that include a parsed
+        grapher object for the grapherConfig field */
+    @observable.ref richDataRows: VariableAnnotationsRow[] | undefined =
+        undefined
+
+    /** Undo step stack - not yet used - TODO: implement Undo/redo */
+    @observable undoSteps: UndoStep[] = []
+    /** Redo step stack - not yet used */
+    @observable redoSteps: UndoStep[] = []
+    // Filter fields and paging offset
+    @observable variableNameFilter: string | undefined = undefined
+    @observable datasetNameFilter: string = ""
+    /** This field stores the offset of what is currently displayed on screen */
+    @observable currentPagingOffset: number = 0
+    /** This field stores the offset that the user requested. E.g. if the user clicks
+        the "Next page" quickly twice then this field will be 2 paging offsets higher
+        than currentPagingOffset until the request to retrieve that data is complete
+        at which point they will be the same */
+    @observable desiredPagingOffset: number = 0
+    // Sorting fields - not yet used - TODO: implement sorting
+    @observable sortByColumn: string = "id"
+    @observable sortByAscending: boolean = false
+    disposers: Disposer[] = []
+    /** Computed property that combines all relevant filter, paging and offset
+        properties into a FetcHVariablesParameter object that is used to construct
+        the query */
+    @computed get dataFetchParameters(): FetchVariablesParameters {
+        const {
+            variableNameFilter,
+            datasetNameFilter,
+            desiredPagingOffset,
+            sortByAscending,
+            sortByColumn,
+        } = this
+        const filterOperations = [
+            searchFieldStringToFilterOperations(
+                variableNameFilter ?? "",
+                new SqlColumnName(SQL_COLUMN_NAME_VARIABLE_NAME)
+            ),
+            searchFieldStringToFilterOperations(
+                datasetNameFilter ?? "",
+                new SqlColumnName(SQL_COLUMN_NAME_DATASET_NAME)
+            ),
+        ].filter((item): item is Operation => item !== undefined)
```

Comment:
nit: there's an `excludeUndefined()` utility that we use in cases like this, it has the type guard too.